buildspec.yml

```diff
@@ -1,30 +1,56 @@
 version: 0.2
 
+env:
+  variables:
+    ENVIRONMENT: "dev"
+    IMAGE_NAME: "dev-swsoc-docker-lambda-base"
+
 phases:
   pre_build:
     commands:
-      - echo Logging in to Amazon ECR...
-      - aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/w5r9l1c8
-      
+      - |
+        echo Logging in to Amazon ECR...
+        aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/w5r9l1c8
+        if git describe --tags --exact-match > /dev/null 2>&1; then
+          echo "This is a tag push event"
+          ENVIRONMENT="prod"
+          IMAGE_NAME="swsoc-docker-lambda-base"
+          GIT_TAG=`git describe --tags --exact-match`
+        elif [[ "${CDK_ENVIRONMENT}" == "PRODUCTION" ]]; then
+          echo "This is a production environment"
+          ENVIRONMENT="prod"
+          IMAGE_NAME="swsoc-docker-lambda-base"
+          GIT_TAG=`date -u +"%Y%m%d%H%M%S"`
+        else
+          echo "This is a development environment"
+          GIT_TAG=`date -u +"%Y%m%d%H%M%S"`
```

Comment:
Does this case also need to define the IMAGE_NAME and ENVIRONMENT?

Reply:
It's already set as the environmental variable as default above on the file:
```
env:
  variables:
    ENVIRONMENT: "dev"
    IMAGE_NAME: "dev-swsoc-docker-lambda-base"
```